augur/filter.py

```diff
@@ -1050,7 +1052,19 @@ def get_groups_for_subsampling(strains, metadata, group_by=None):
                     })
                 # month = (year, month)
                 metadata['month'] = list(zip(metadata['year'], metadata['month']))
-            # TODO: support group by day
+            if 'week' in generated_columns_requested:
+                # skip ambiguous days
+                df_skip = metadata[metadata['day'].isnull()]
+                metadata.dropna(subset=['day'], inplace=True)
+                for strain in df_skip.index:
+                    skipped_strains.append({
+                        "strain": strain,
+                        "filter": "skip_group_by_with_ambiguous_day",
+                        "kwargs": "",
+                    })
+                week_of_year = metadata.apply(lambda row: get_week(row['year'], row['month'], row['day']), axis=1)
+                # week = (year, week)
+                metadata['week'] = list(zip(metadata['year'], week_of_year))
```

Comment:
This has a bug – the raw `year` should not be combined with the ISO week.

```sh
cat >metadata.tsv <<~~
strain	date
SEQ1	2003-01-01
SEQ2	2003-01-02
SEQ3	2003-12-29
SEQ4	2003-12-30
~~

augur filter \
   --metadata metadata.tsv \
   --group-by week \
   --sequences-per-group 1 \
   --subsample-seed 0 \
   --output-metadata out.tsv
# 3 strains were dropped during filtering
# 	3 of these were dropped because of subsampling criteria
# 1 strains passed all filters

cat out.tsv
# strain	date
# SEQ1	2003-01-01
```

Reply:
Fixed the above bug with [this force-push](https://github.com/nextstrain/augur/compare/c4df4607260a315eea6b3584e161c61a04a174cb..6c21ae279053960574148f11fb173f14c80e446f).

Found another similar bug and fixed with [this force-push](https://github.com/nextstrain/augur/compare/6c21ae279053960574148f11fb173f14c80e446f..a6ec84aef05cc3a944acb6dfe5a4ec54a2bd99bb).